spec/support/examples_group/shared_examples_for_aggregation_mixin.rb

```diff
@@ -0,0 +1,54 @@
+shared_examples_for "AggregationMixin" do |through|
+  context "includes AggregationMixin" do
+    include Spec::Support::ArelHelper
+
+    let(:hardware) do
+      FactoryBot.create(:hardware,
+                        :cpu2x2,
+                        :ram1GB,
+                        :cpu_speed     => 2_999,
+                        :disk_capacity => 40,
+                        :memory_mb     => 2048)
+    end
+
+    let(:ems) { FactoryBot.create(:ext_management_system) }
+    let!(:vm) { FactoryBot.create(:vm, :hardware => hardware, :ext_management_system => ems) }
+    let!(:vm1) { FactoryBot.create(:vm, :hardware => hardware, :ext_management_system => ems) }
+    let!(:host) { FactoryBot.create(:host, :hardware => hardware, :ext_management_system => ems) }
+    let!(:host1) { FactoryBot.create(:host, :hardware => hardware, :ext_management_system => ems) }
+    let(:object) { FactoryBot.create(described_class.to_s.underscore.to_sym) }
+
+    before do
+      if through == "ext_management_systems"
+        object.ext_management_systems << ems
+      elsif through == "computer_systems"
+        object.computer_system_hardwares << [hardware]
+      elsif described_class == Host
+        object.host_hardwares << [hardware]
+        object.vms_and_templates << [vm, vm1]
+      else
+        object.hosts << [host, host1]
+        object.vms_and_templates << [vm, vm1]
+      end
+    end
+
+    describe "calculates single object" do
+      context "host" do
```

Comment:
With the comment below, you probably don't need the context if you don't want.

Reply:
yes but i do want, sorry

Reply:
fair enough.  resolved.